Manatee.Json/Schema/Keywords/AdditionalPropertiesKeyword.cs

```diff
@@ -128,7 +128,10 @@ public SchemaValidationResults Validate(SchemaValidationContext context)
 				{
 					failedProperties.Add(kvp.Key);
 				}
-				context.EvaluatedPropertyNames.UnionWith(newContext.EvaluatedPropertyNames);
+
+				if (newContext.EvaluatedPropertyNames.Count > context.EvaluatedPropertyNames.Count)
```

Comment:
What if the `newContext` has a different set of property names, but is the same size?